pcap/pcapio/ngread.go

```diff
@@ -0,0 +1,310 @@
+// The code in this source file is derived from
+// https://github.com/google/gopacket/blob/master/pcapgo/ngread.go
+// as of February 2020 and is covered by the copyright below.
+// The changes are covered by the copyright and license in the
+// LICENSE file in the root directory of this repository.
+
+// Copyright 2018 The GoPacket Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package pcapio
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/pkg/peeker"
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+)
+
+var ErrCorruptPcap = errors.New("bad pcap file")
+
+const PacketBlockHeaderLen = 28
+
+// NgReader wraps an underlying bufio.NgReader to read packet data in pcapng.
+type NgReader struct {
+	*peeker.Reader
+	ifaces        []NgInterface
+	currentOption ngOption
+	first         []byte
+	ci            gopacket.CaptureInfo
+	bigEndian     bool
+	offset        uint64
+}
+
+// NewNgReader initializes a new writer, reads the first section header,
+// and if necessary according to the options the first interface.
+func NewNgReader(r io.Reader) (*NgReader, error) {
+	ret := &NgReader{
+		Reader: peeker.NewReader(r, 32*1024, 1024*1024),
+	}
+
+	//pcapng _must_ start with a section header
+	typ, block, err := ret.readBlock()
```

Comment:
If this pcap was recorded with written with bigEndian byte order (the default is littleEndian) I believe this going to get return an error: "pcap-ng trailer length mismatch" (Same goes for `Read()`. I believe you're going to need to read the block type bits, and if its a section header run this through `parseSectionHeader`, before  calling `readBlock`.

Reply:
Will fix it!

Reply:
good catch